enumerables.rb

```diff
@@ -0,0 +1,131 @@
+module Enumerable
+  def my_each
+    size = count - 1
+    counter = 0
+    items = to_a
+    while counter <= size
+      if is_a? Range
+        items.length >= 0 && yield(items[counter])
+      elsif is_a? Hash
+        yield(items[counter][0], items[counter][1])
+      elsif !block_given?
+        return self
+      else
+        yield(self[counter])
+      end
+      counter += 1
+    end
+    self
+  end
+
+  def my_each_with_index()
+    counter = 0
+    size = length - 1
+    return self unless block_given?
+
+    while counter <= size
+      yield(self[counter], counter)
+      counter += 1
+    end
+    self
+  end
+
+  def my_select
+    return self unless block_given?
+
+    required = []
+
+    my_each do |r|
+      required.push(yield(r))
+    end
+    required
+  end
```

Comment:
Could you test with ruby's `select`? This method should return an array contained in the initial enumerable that satify the condition in the block given. See the example below.

```ruby
[1,2,3,7,8,9].select{|i| i <  5}
#=> [1, 2, 3]

[1,2,3,7,8,9].my_select{|i| i <  5}
#=> [true, true, true, false, false, false]
```

HINT

```suggestion
  def my_select
    return self.to_enum unless block_given?
    
    required = []

    my_each do |r|
      required.push(r) if yield(r)
    end
    required
  end
```